### PR TITLE
Stop showing a commit button when nothing is new. I tracked down vari…

### PR DESCRIPTION
**Pull request by boutell**
…ous performance issues with the editable route and in the process of eliminating duplicate ids that led to poor performance, this bug was suppressed. Also I added an apostrophe-workflow:resolve-join-ids task for cleaning up the consequences of earlier bugs where not all types of joins got resolved.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -661,6 +661,7 @@ module.exports = function(self, options) {
           self.deleteExcludedProperties(_draft);
           self.deleteExcludedProperties(_live);
           if (!_.isEqual(_draft, _live)) {
+            // console.log('***', JSON.stringify(_draft, null, '  '), JSON.stringify(_live, null, '  '));
```
Comment (bobclewell) [inline lib/api.js]: Could probably remove this debugging output.

File: lib/api.js [hunk 2]
```diff
@@ -972,6 +978,9 @@ module.exports = function(self, options) {
       // Modified. Could also be moved, so don't return
       if (!_.isEqual(value, toObjects.byId[value._id])) {
         if (description) {
+          // console.log(toObjects.dotPaths[value._id]);
+          // console.log(JSON.stringify(value, null, '  '));
+          // console.log(JSON.stringify(toObjects.byId[value._id], null, '  '));
```
Comment (bobclewell) [inline lib/api.js]: More debug comments here and above.

File: lib/api.js [hunk 3]
```diff
@@ -634,16 +635,15 @@ module.exports = function(self, options) {
...
-
     function diff(callback) {
-      return async.eachSeries(related, function(doc, callback) {
+      var ids = _.map(related, function(item) {
+        return item._id || (item.item && item.item._id);
+      });
+      ids = _.uniq(ids);
+      ids = _.filter(ids, function(id) {
+        return !!id;
+      });
```
Comment (bobclewell) [inline lib/api.js]: Seems like a opportunity to chain these methods?

```
var ids = _.map(related, function(item) {
    return item._id || (item.item && item.item._id);
  })
  .uniq(ids)
  .filter(ids, function(id) {
    return !!id;
  });
```

File: lib/tasks.js [hunk 4]
```diff
@@ -133,6 +140,36 @@ module.exports = function(self, options) {
     }
   };
 
+  self.resolveJoinIdsTask = function(apos, argv, callback) {
+    var req = self.apos.tasks.getReq();
+    
+    return async.series([
```
Comment (bobclewell) [inline lib/tasks.js]: 👍

File: lib/api.js (hunk 2 above)
Comment (boutell) [reply lib/api.js]: These debug comments are likely to be useful a few more times before we see the last of debugging this algorithm, so I'm happy with leaving them present but commented like this.

File: lib/api.js (hunk 3 above)
Comment (boutell) [reply lib/api.js]: Actually, if we're going to do that, then we have to use _.chain(related) to start the chain and _.value() at the end of the chain so lodash knows where to stop. Or we can use _(related) as a shorthand. I think the existing code is clear and correct, so I'm going to leave it in place for now, but we should look at using lodash chaining more in future.

**Pull request merged**